modules/ejbca-ejb-cli/src/org/ejbca/ui/cli/ca/CaImportCertDirCommand.java

```diff
@@ -214,6 +233,56 @@ public CommandResult execute(final ParameterContainer parameters) {
 
 
             for (final File file : files) {
+                
+                // Check if revocation details are to be derived from the filename. Only do this if status is REVOKED
+                if ( (status == CertificateConstants.CERT_REVOKED) &&  bRevokeDetailsInFilename) {
+                    // Find the revocation details from the filename. The details are separated with an exclamation (!) character.
+                    final String[] sa = file.getName().split("!");
+                    if (sa.length <3) {
+                        log.error("ERROR: The revocation details are not found in filename '"+file.getName()+"'. Ignoring this file.");
+                        continue;
+                   } else {
+                        // Process the REASON from 2nd last string in array
+                        String sRevCode = sa[ sa.length-2 ].toUpperCase();
+                        // Check if using a code value
+                        try {
+                            final int iRevCode = Integer.parseInt(sRevCode);
+                            revocationReason = RevocationReasons.getFromDatabaseValue(iRevCode);
+                            if(revocationReason == null) {
+                                log.error("ERROR: '" + iRevCode + "' is not a valid revocation reason code. Ignoring this file.");
+                                continue;
+                            }
+                            
+                        } catch (NumberFormatException e) {
+                            // Not an integer, must be the full text
+                            // Correct the string value to suit RevocationReason
+                            if ( sRevCode.equals("KEYCOMPROMISE")) sRevCode = "KEY_COMPROMISE";
+                            if ( sRevCode.equals("CACOMPROMISE")) sRevCode = "CA_COMPROMISE";
+                            if ( sRevCode.equals("AFFILIATIONCHANGED")) sRevCode = "AFFILIATION_CHANGED";
+                            if ( sRevCode.equals("CESSATIONOFOPERATION")) sRevCode = "CESSATION_OF_OPERATION";
+                            if ( sRevCode.equals("CERTIFICATEHOLD")) sRevCode = "CERTIFICATE_HOLD";
+                            if ( sRevCode.equals("PRIVILEGESWITHDRAWN")) sRevCode = "PRIVILEGES_WITHDRAWN";
+                            if ( sRevCode.equals("CERTIFICATEHOLD")) sRevCode = "AA_COMPROMISE";
+                                                             
+                            revocationReason = RevocationReasons.getFromCliValue(sRevCode.toUpperCase());
+                            if(revocationReason == null) {
+                                log.error("ERROR: '" + sRevCode + "' is not a valid revocation reason. Ignoring this file.");
```

Comment:
Message should print what file is ignored, otherwise it's impossible to know.

Reply:
OK